### PR TITLE
DIV-2455: No JWT

### PR DESCRIPTION
**Pull request by david-gimelle**
# Description

Div-2455 and Div-2508 - Refactoring in order to stop to take user details from the JWT token and take this information from idam /details endpoints

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Regression testing only

# Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/draftservice/service/DraftsService.java [hunk 1]
```diff
@@ -25,6 +26,9 @@
     @Autowired
     private DraftStoreClient client;
 
+    @Autowired
+    private UserService userService;
```
Comment (dnenkov) [inline src/main/java/uk/gov/hmcts/reform/divorce/draftservice/service/DraftsService.java]: You are not using the user service in this class. You are just passing the `jwt` to the key factory

File: src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/UpdateService.java [hunk 2]
```diff
@@ -35,12 +37,17 @@
     @Autowired
     private PetitionValidatorService petitionValidatorService;
 
+    @Autowired
+    UserService userService;
```
Comment (dnenkov) [inline src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/UpdateService.java]: Can this `private`

Comment (david-gimelle) [reply src/main/java/uk/gov/hmcts/reform/divorce/transformservice/service/UpdateService.java]: Done

File: src/main/java/uk/gov/hmcts/reform/divorce/draftservice/service/DraftsService.java (hunk 1 above)
Comment (david-gimelle) [reply src/main/java/uk/gov/hmcts/reform/divorce/draftservice/service/DraftsService.java]: done

Comment (dnenkov) [reply src/main/java/uk/gov/hmcts/reform/divorce/draftservice/service/DraftsService.java]: Sorry, @david-gimelle  can you please point me to the commit where you have fixed this. I can't see the change for some reason.

File: docker-compose.yml [hunk 3]
```diff
@@ -358,6 +358,7 @@ services:
       - AUTH_PROVIDER_SERVICE_CLIENT_BASEURL=http://service-auth-provider-api:8080
       - DRAFT_STORE_API_BASEURL=http://draft-store-api:8800
       - PDF_GENERATOR_BASE_URL=http://div-document-generator:4015
+      - IDAM_API_BASEURL=http://localhost:4501
```
Comment (dnenkov) [inline docker-compose.yml]: This has to be `http://idam-api:8080`

Comment (david-gimelle) [reply docker-compose.yml]: Done

**Pull request merged**